src/knot_protocol.c

```diff
@@ -172,21 +159,20 @@ int knot_type_id_is_generic(uint16_t type_id)
 
 int knot_schema_is_valid(uint16_t type_id, uint8_t value_type, uint8_t unit)
 {
-	/* int/float/bool/raw ? */
+	/* int/float/bool/raw/int64/uint/uint64 ? */
 	if (knot_value_type_is_valid(value_type) == 0) {
 
 		/* Verify basic type IDs */
 		if (knot_type_id_is_basic(type_id) == 0) {
-			if (basic_types[type_id].value_type == value_type &&
-			    basic_types[type_id].unit_mask & UNIT_MASK(unit))
+			if (basic_types[type_id] & UNIT_MASK(unit))
 				return 0;
 
 		/* Verify logic type IDs */
 		} else if (knot_type_id_is_logic(type_id) == 0) {
-			if (logic_types[TYPE_ID_MASK(type_id)] == value_type)
+			if (unit == KNOT_UNIT_NOT_APPLICABLE)
 				return 0;
 		} else if (knot_type_id_is_generic(type_id) == 0) {
-			if (generic_types[TYPE_ID_MASK(type_id)] == value_type)
+			if (unit == KNOT_UNIT_NOT_APPLICABLE)
```

Comment:
Are generic values covering analogic values? If so, does it make sense to limit the unit to be not applicable?

Reply:
Yes. I am not sure. I tried to follow the same pattern described in [KNoT Documentation](https://knot-devel.cesar.org.br/doc/thing/unit-type-value.html) and implemented in [KNoT BabelTower schema rules](https://github.com/CESARBR/knot-babeltower/blob/b4a9de80f482495740733a5a0f71daf43a54201e/pkg/thing/interactors/update_schema.go#L47). What do you think @vitbaq?

Reply:
@vitbaq what do you think?

Reply:
Because each unit is invariably tied to some other Type Id, I don't think it would make sense to allow other units. That being said, it's not clear to me what's the use case for Type Id Analog.

Reply:
I suggest changing the name from analog to generic.
KNOT_TYPE_ID_ANALOG -> KNOT_TYPE_ID_GENERIC

If you use an analogic sensor you can choose any type_id, including the generic type.
The generic type represents all other types that are not mapped by the protocol. And i think that the only unit that can be used is KNOT_UNIT_NOT_APPLICABLE. Or we can create a new unit: KNOT_UNIT_GENERIC

Reply:
It's now making sense for me. I think we can go with `KNOT_UNIT_NOT_APPLICABLE`.

Reply:
I add a commit replacing the KNOT_TYPE_ID_ANALOG with KNOT_TYPE_ID_GENERIC and adding the KNOT_UNIT_GENERIC.